tests/CMakeLists.txt

```diff
@@ -14,7 +14,7 @@ function(add_iota_test test_src test_name)
     target_link_libraries(${test_name} PRIVATE iota_core unity)
   elseif(${test_src} MATCHES "^crypto")
     add_dependencies(${test_name} iota_crypto)
-    target_link_libraries(${test_name} PRIVATE iota_crypto unity)
+    target_link_libraries(${test_name} PRIVATE iota_crypto iota_core unity)
```

Comment:
crypto provides service for the core layer, therefore iota_crypto unit tests should be standalone, not consume iota_core.
https://iota-c-client.readthedocs.io/en/latest/client_intro.html#c-client-library-diagram

Reply:
I was using iota_core mainly for the hex_2_bin method in bute_buffer.c. I removed iota_core dependency and added hex_2_bin as static function in test_crypto.c